lib/_http_agent.js

```diff
@@ -332,6 +317,25 @@ Agent.prototype.destroy = function destroy() {
   }
 };
 
+function handleSocketCreatingError(req, tellReq) {
+  return function handleSocketCreatingErrorInner(err, socket) {
+    function emitErr() {
+      req.emit('error', err);
+    }
+
+    if (!err) {
+      return tellReq ? req.onSocket(socket) : socket.emit('free');
+    }
+
+    const asyncId = socket && socket._handle && socket._handle.getAsyncId();
+    if (asyncId) {
+      nextTick(asyncId, emitErr);
```

Comment:
You can pass `null` here and the internal `nextTick()` will grab the correct id.